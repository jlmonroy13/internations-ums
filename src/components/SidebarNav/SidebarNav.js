import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

function SidebarNav({ url, logout, isNavOpen }) {
  const mobileClass = isNavOpen ? 'sidebar-nav__mobile' : '';
  return (
    <nav className={`sidebar-nav ${mobileClass}`}>
      <Link className={`sidebar-nav__btn ${url === 'users' ? 'is-active' : ''}`} to="/users" >
        <span className="sidebar-nav__btn-icon icon icon--users" />
        <span className="sidebar-nav__btn-text">Users</span>
      </Link>
      <Link className={`sidebar-nav__btn ${url === 'groups' ? 'is-active' : ''}`} to="/groups">
        <span className="sidebar-nav__btn-icon icon icon--group" />
        <span className="sidebar-nav__btn-text">Groups</span>
      </Link>
      <div className="sidebar-nav__btn" onClick={logout}>
        <span className="sidebar-nav__btn-icon icon icon--log-out" />
        <span className="sidebar-nav__btn-text">Log out</span>
      </div>
    </nav>
  );
}

SidebarNav.propTypes = {
  url: PropTypes.string.isRequired,
  logout: PropTypes.func.isRequired,
  isNavOpen: PropTypes.bool.isRequired,
};

export default SidebarNav;
