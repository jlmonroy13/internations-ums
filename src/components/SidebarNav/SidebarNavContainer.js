import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SidebarNav from './SidebarNav';
import { logout } from '../../actions/account';

function mapStateToProps(state) {
  const { routing: { locationBeforeTransitions: { pathname } }, events: { isNavOpen } } = state;
  const url = pathname.substring(1);

  return {
    url,
    isNavOpen,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ logout }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(SidebarNav);
