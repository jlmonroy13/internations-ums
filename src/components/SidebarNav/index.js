import SidebarNavContainer from './SidebarNavContainer';

export default SidebarNavContainer;
