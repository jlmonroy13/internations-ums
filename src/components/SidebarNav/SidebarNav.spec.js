import React from 'react';
import renderer from 'react-test-renderer';
import SidebarNav from './SidebarNav';

describe('<SidebarNav />', () => {
  it('should render the sidebar nav component', () => {
    const logout = jest.fn();
    const wrapper = renderer.create(<SidebarNav url="" logout={logout} isNavOpen={true}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
