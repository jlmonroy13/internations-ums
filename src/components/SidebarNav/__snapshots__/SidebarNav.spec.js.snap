// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarNav /> should render the sidebar nav component 1`] = `
<nav
  className="sidebar-nav sidebar-nav__mobile"
>
  <a
    className="sidebar-nav__btn "
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="sidebar-nav__btn-icon icon icon--users"
    />
    <span
      className="sidebar-nav__btn-text"
    >
      Users
    </span>
  </a>
  <a
    className="sidebar-nav__btn "
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="sidebar-nav__btn-icon icon icon--group"
    />
    <span
      className="sidebar-nav__btn-text"
    >
      Groups
    </span>
  </a>
  <div
    className="sidebar-nav__btn"
    onClick={[Function]}
  >
    <span
      className="sidebar-nav__btn-icon icon icon--log-out"
    />
    <span
      className="sidebar-nav__btn-text"
    >
      Log out
    </span>
  </div>
</nav>
`;
